language-plutus-core/exe/Main.hs

```diff
@@ -250,29 +253,37 @@ getProg inp fmt =
                return (fakeAlexPosn <$ plc)  -- Adjust the return type to ParsedProgram
                    where fakeAlexPosn = PLC.AlexPn 0 0 0
 
+
 ---------------- Typechecking ----------------
 
 runTypecheck :: TypecheckOptions -> IO ()
-runTypecheck (TypecheckOptions inp fmt) = do
-    prog <- getProg inp fmt
-    let cfg = PLC.defConfig
-    case PLC.runQuoteT $ PLC.typecheckPipeline cfg prog of
-      Left (e :: PlcParserError) -> do
-            T.putStrLn $ PLC.displayPlcDef e
-            exitFailure
-      Right ty -> do
-            T.putStrLn $ PLC.displayPlcDef ty
-            exitSuccess
+runTypecheck (TypecheckOptions inp fmt) =
+    case PLC.runQuoteT $ PLC.getStringBuiltinTypes () of
+      Left (e :: PLC.Error PLC.DefaultUni ()) -> do
+          T.putStrLn $ PLC.displayPlcDef e
+          exitFailure
+      Right dyntypes -> do
+          let cfg = PLC.TypeCheckConfig dyntypes
+          prog <- getProg inp fmt
+          case PLC.runQuoteT $ PLC.typecheckPipeline cfg prog of
+            Left (e :: PlcParserError) -> do
+                          T.putStrLn $ PLC.displayPlcDef e
+                          exitFailure
+            Right ty -> do
+              T.putStrLn $ PLC.displayPlcDef ty
+              exitSuccess
 
 
 ---------------- Evaluation ----------------
 
+
 runEval :: EvalOptions -> IO ()
 runEval (EvalOptions inp mode fmt) = do
   prog <- getProg inp fmt
-  let evalFn = case mode of
+  let meanings = PLC.getStringBuiltinMeanings @ (PLC.Term PLC.TyName PLC.Name PLC.DefaultUni PLC.ExMemory)
```

Comment:
If you omit the explicit type application, the code will break?

Reply:
Hmmm.  Apparently not.  I'm sure I needed it before though.  Maybe I needed that in an earlier version of the typechecking code and just pasted it into the evaluation code.